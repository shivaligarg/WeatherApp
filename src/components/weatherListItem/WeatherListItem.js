import React from 'react';
import { Image } from 'react-native';
import { ListItem} from 'react-native-elements';
import { stringFile } from '../../utils/strings';
import { IMAGE_API_URL } from '../../utils/url';
export const WeatherListItem =(props) =>{
    const {item}=props
     return (
        <ListItem bottomDivider>
        <ListItem.Content >
            {item.dt_txt&&<ListItem.Title style={{fontWeight:'bold',fontSize:20}}>{item.dt_txt}</ListItem.Title>}
            <ListItem.Content style={{flexDirection:"row" ,marginTop:20}}>
            {item.main&&(item.main.temp||item.main.temp_min||item.main.temp_max)?<ListItem.Content >
            <ListItem.Title >{stringFile.list.temp}</ListItem.Title>
            {item.main.temp&&<ListItem.Subtitle>{stringFile.list.tempAt+item.main.temp.toFixed(1)}</ListItem.Subtitle>}
            {item.main.temp_min&&<ListItem.Subtitle>{stringFile.list.minTemp+item.main.temp_min.toFixed(1)}</ListItem.Subtitle>}
            {item.main.temp_max&&<ListItem.Subtitle>{stringFile.list.maxTemp+item.main.temp_max.toFixed(1)}</ListItem.Subtitle>}
            </ListItem.Content>:null}
            {item.weather&&item.weather.length>0&&item.weather[0].main?
            <ListItem.Content >
            <ListItem.Title >{stringFile.list.weather}</ListItem.Title>
            <ListItem.Content style={{flexDirection:"row",alignItems:"center",flexWrap:"wrap"}}>
            <Image source={{uri:IMAGE_API_URL.replace("{0}",item.weather[0].icon)}} style={{width:35,height:35}}/>
            <ListItem.Subtitle>{item.weather[0].main}</ListItem.Subtitle>
            </ListItem.Content>
            </ListItem.Content>:null}
            </ListItem.Content>
            </ListItem.Content>
        </ListItem>
    );
}
