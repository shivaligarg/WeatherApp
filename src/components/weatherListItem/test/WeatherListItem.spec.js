import React from 'react';
import {shallow} from 'enzyme';
import { WeatherListItem } from '../WeatherListItem';
import { ListItem } from 'react-native-elements';
import { Image } from 'react-native';

describe('Weather List Item Check',()=>{
    let wrapper;
    let itemProp= {
        "dt": 1603962000,
        "main": {
          "temp": 290.04,
          "feels_like": 288.42,
          "temp_min": 290.04,
          "temp_max": 292.03,
          "pressure": 1017,
          "sea_level": 1017,
          "grnd_level": 997,
          "humidity": 48,
          "temp_kf": -1.99
        },
        "weather": [
          {
            "id": 800,
            "main": "Clear",
            "description": "clear sky",
            "icon": "01n"
          }
        ],
        "clouds": {
          "all": 1
        },
        "wind": {
          "speed": 0.95,
          "deg": 48
        },
        "visibility": 10000,
        "pop": 0,
        "sys": {
          "pod": "n"
        },
        "dt_txt": "2020-10-29 09:00:00"
      }
    beforeEach(()=>{
        wrapper=shallow(<WeatherListItem item={itemProp}/>);
    });

    it('Weather List Item Snapshot',()=>{
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(ListItem.Title).length).toBe(3);
        expect(wrapper.find(ListItem.Subtitle).length).toBe(4);
        expect(wrapper.find(ListItem.Content).length).toBe(5);       
        expect(wrapper.find(Image).length).toBe(1);    
    });
    it('Weather List Item Snapshot',()=>{
        wrapper.setProps({item:{
            "dt": 1603962000,
            "main": {
            },
            "weather": [
            ],
            "clouds": {
              "all": 1
            },
            "wind": {
              "speed": 0.95,
              "deg": 48
            },
            "visibility": 10000,
            "pop": 0,
            "sys": {
              "pod": "n"
            },
            "dt_txt": ""
          }});
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(ListItem.Title).length).toBe(0);
        expect(wrapper.find(ListItem.Subtitle).length).toBe(0);
        expect(wrapper.find(ListItem.Content).length).toBe(2);     
        expect(wrapper.find(Image).length).toBe(0);  
    });
})