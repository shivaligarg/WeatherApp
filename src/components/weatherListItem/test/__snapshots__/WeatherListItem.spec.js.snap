// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather List Item Check Weather List Item Snapshot 1`] = `ShallowWrapper {}`;

exports[`Weather List Item Check Weather List Item Snapshot 2`] = `ShallowWrapper {}`;
